src/gui/AutomationPatternView.cpp

```diff
@@ -340,15 +340,11 @@ void AutomationPatternView::paintEvent( QPaintEvent * )
 	const int lineSize = 3;
 	p.setPen( c.darker( 300 ) );
 
-	for( tact_t t = 1; t < m_pat->timeMapLength().getTact(); ++t )
+	for( tact_t t = 1; t < width() - TCO_BORDER_WIDTH; ++t )
 	{
 		const int tx = x_base + static_cast<int>( ppt * t ) - 2;
-		if( tx < ( width() - TCO_BORDER_WIDTH * 2 ) )
-		{
-			p.drawLine( tx, TCO_BORDER_WIDTH, tx, TCO_BORDER_WIDTH + lineSize );
-			p.drawLine( tx,	rect().bottom() - ( lineSize + TCO_BORDER_WIDTH ),
-						tx, rect().bottom() - TCO_BORDER_WIDTH );
-		}
+		p.drawLine( tx, TCO_BORDER_WIDTH, tx, TCO_BORDER_WIDTH + lineSize );
+		p.drawLine( tx,	rect().bottom() - ( lineSize + TCO_BORDER_WIDTH ), tx, rect().bottom() - TCO_BORDER_WIDTH );
```

Comment:
The old format of this line, broken into two lines, is more along the recommended 80 character lines width in the [coding conventions](https://github.com/LMMS/lmms/wiki/Coding-conventions). I suggest you change this back to the way it was ( but with `tx, ` 5 tabs in... ) .